standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -543,10 +557,30 @@ static void prewarm(RawStore rawStore) {
                 tableColStats = rawStore.getTableColumnStatistics(catName, dbName, tblName, colNames, CacheUtils.HIVE_ENGINE);
                 Deadline.stopTimer();
               }
+              Deadline.startTimer("getPrimaryKeys");
+              primaryKeys = rawStore.getPrimaryKeys(catName, dbName, tblName);
+              Deadline.stopTimer();
+              cacheObjects.setPrimaryKeys(primaryKeys);
+
+              Deadline.startTimer("getForeignKeys");
+              foreignKeys = rawStore.getForeignKeys(catName, null, null, dbName, tblName);
```

Comment:
Then should we would need store foreign key mappings against parentDb and table for quick access (otherwise we will be scanning all the db/tables in cache)? 

And this also means we will be keeping two copies, one with parent table and another with foreign table.

Reply:
We can have the foreignkeys kept under foreignkey table wrapper but the reference such as foreignkey db/table and key name in parent table wrapper. It will be useful when getForeignKeys is called with null for foreign db/tbl. I just want to confirm if this a frequent call with null as input. If yes, then let's do it, if not ignore this comment.

Reply:
Current usages in code of `getForeignKeys`  contains null only for parentDb/table, foreignDb/table are always being populated. So let's skip it as you mentioned.